include/net/dsa.h

```diff
@@ -0,0 +1,174 @@
+/** @file
+ @brief DSA definitions and handlers
+
+ */
+
+/*
+ * Copyright (c) 2020 DENX Software Engineering GmbH
+ *               Lukasz Majewski <lukma@denx.de>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_DSA_H_
+#define ZEPHYR_INCLUDE_NET_DSA_H_
+
+#include <device.h>
+#include <net/net_if.h>
+
+/**
+ * @brief LLDP definitions and helpers
+ * @defgroup lldp Link Layer Discovery Protocol definitions and helpers
+ * @ingroup networking
+ * @{
+ */
+
+#define NET_DSA_PORT_MAX_COUNT 8
+
+#define DSA_NET_DEVICE_INIT(dev_name, drv_name, init_fn, pm_control_fn,	\
+			    data, cfg_info, prio, api, mtu, port_num)	  \
+	DEVICE_DEFINE(dev_name, drv_name, init_fn, pm_control_fn, data,	\
+		      cfg_info, POST_KERNEL, prio, api);		\
+	NET_L2_DATA_INIT(dev_name, 0, NET_L2_GET_CTX_TYPE(ETHERNET_L2)); \
+	NET_IF_INIT(dev_name, 0, ETHERNET_L2, mtu, port_num)
+
+#define DSA_STATUS_TICK_MS K_MSEC(1000)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief DSA function to adjust packet
```

Comment:
I do not understand the description, which says the function will adjust the packet, but the function name is dsa_xmit_pkt(). Can't we just call this dsa_adjust_pkt() then, or change the description to say that function will transmit the pkt?

Reply:
This function - i.e. dsa_xmit_pkt() - is supposed to be used to implement the tail tagging. I will re-check the naming and update the description accordingly.